source/server-discovery-and-monitoring/tests/unified/interruptInUse-pool-clear.yml

```diff
@@ -0,0 +1,354 @@
+---
+description: interruptInUse
+
+schemaVersion: "1.11"
+
+runOnRequirements:
+  # failCommand appName requirements
+  - minServerVersion: "4.9"
+    serverless: forbid
+    topologies: [ replicaset, sharded ]
+
+createEntities:
+  - client:
+      id: &setupClient setupClient
+      useMultipleMongoses: false
+
+initialData: &initialData
+  - collectionName: &collectionName interruptInUse
+    databaseName: &databaseName sdam-tests
+    documents: []
+
+tests:
+  - description: Connection pool clear uses interruptInUseConnections=true after monitor timeout
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                useMultipleMongoses: false
+                observeEvents:
+                  - poolClearedEvent
+                  - connectionClosedEvent
+                  - commandStartedEvent
+                  - commandSucceededEvent
+                  - commandFailedEvent
+                  - connectionCheckedOutEvent
+                  - connectionCheckedInEvent
+                uriOptions:
+                  connectTimeoutMS: 500
+                  heartbeatFrequencyMS: 500
+                  appname: interruptInUse
+                  retryReads: false
+                  minPoolSize: 0
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+            - thread:
+                id: &thread1 thread1
+      - name: insertOne
+        object: *collection
+        arguments:
+          document: { _id: 1 }
+      # simulate a long-running query
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          thread: *thread1
+          operation:
+            name: find
+            object: *collection
+            arguments:
+              filter:
+                $where : sleep(2000) || true
+            expectError:
+              isError: true
+      - name: waitForEvent
```

Comment:
it looks like this test is a bit flaky. Since previous `runOnThread` ensures that thread is started, but not that underlying operation has actually been launched. So there was a chance that failpoint is triggered before find is started

Reply:
> So there was a chance that failpoint is triggered before find is started

Would increasing "times" from 2 to 4 help here?

Reply:
The failpoint being triggered before the find has started should be okay, so long as the monitor check doesn't time out before the find begins. In the quickest case, the monitor can take 1000ms to time out after the failpoint is enabled (a check happens to start immediately after the failpoint is enabled), and the find should certainly take less than 1000ms to begin executing, considering the pool has already been populated with a connection. If we wait for the find to start before enabling the failpoint, we instead are concerned with the find completing before the monitor times out (in longest case, monitor could take 1500ms after the failpoint is enabled to time out, which should still be okay assuming it takes less than 500ms to enable the failpoint).

Can you elaborate more on the events you were seeing when it was failing?

>Would increasing "times" from 2 to 4 help here?

This could possibly be the cause, though I imagine it would be pretty rare. If the RTT monitor hits the failpoint, times out, and then hits it again, it could prevent the server monitor from ever triggering it if the monitor happened to start a check right before the failpoint was enabled.

Reply:
>Can you elaborate more on the events you were seeing when it was failing?

I see than in some rare cases no error happens at all

Reply:
In that case, @ShaneHarvey's suggestion of upping to `times: 4` may help. Did you see any difference after adding the `waitForEvent`?

Reply:
>Would increasing "times" from 2 to 4 help here?

yep, it works too, done

Reply:
done